File: 0541-reverse-string-ii/0541-reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i += 2 * k) {
                int left = i;
                int right = Math.min(i + k - 1, chars.length - 1);

                while (left < right) {
                    char tmp = chars[left];
                    chars[left++] = chars[right];
                    chars[right--] = tmp;
                }
        }

        return new String(chars);
    }
}

File: 0541-reverse-string-ii/0541-reverse-string-ii.kt
class Solution {
    fun reverseStr(s: String, k: Int): String {
        val chars = s.toCharArray()
        var i = 0
        while (i < chars.size) {
            var left = i
            var right = minOf(i + k - 1, chars.size - 1)

            while (left < right) {
                chars[left] = chars[right].also { chars[right] = chars[left] }
                left++
                right--
            }
            
            i += 2 * k
        }

        return String(chars)
    }
}